konduit-serving-test/src/test/java/ai/konduit/serving/util/SchemaTypeUtilsTest.java

```diff
@@ -22,16 +22,200 @@
+                                    new JsonObject().put("type",SchemaType.values()[i].name())));
+                    break;
+
+                default:
+                    throw new UnsupportedOperationException("Type " + SchemaType.values()[i].name() + " is unsupported!");
+
+            }
+
+        }
+
+        Schema assertionSchema = schemaBuilder.build();
+        Schema schema1 = SchemaTypeUtils.schemaFromDynamicSchemaDefinition(schema);
+        assertEquals(assertionSchema,schema1);
+
+    }
+
+    @Test
+    public void testTypeMappingsForSchema() {
+        Schema.Builder schemaBuilder = new Schema.Builder();
+        schemaBuilder.addColumnInteger("int");
+        schemaBuilder.addColumnLong("long");
+        schemaBuilder.addColumnNDArray("ndarray",new long[]{1,1});
+        schemaBuilder.addColumnString("string");
+        schemaBuilder.addColumnCategorical("categorical","cat1","cat2");
+        schemaBuilder.addColumnFloat("float");
+        schemaBuilder.addColumnDouble("double");
+        schemaBuilder.addColumnBoolean("boolean");
+
+        Schema convert = schemaBuilder.build();
+        final Map<String, SchemaType> result = SchemaTypeUtils.typeMappingsForSchema(convert);
+
+    }
+
+    @Test
+    public void testToSchema() {
+        SchemaType[] values = SchemaType.values();
+        Schema.Builder schemaBuilder = new Schema.Builder();
+
+        final List<String> names = new ArrayList<>();
+        for(SchemaType value : values) {
+            names.add(value.name());
+            switch(value) {
+                case NDArray:
+                    schemaBuilder.addColumnNDArray(value.name(),new long[]{1,1});
+                    break;
+                case Boolean:
+                    schemaBuilder.addColumnBoolean(value.name());
+                     break;
+                case Float:
+                    schemaBuilder.addColumnFloat(value.name());
+                    break;
+                case Double:
+                    schemaBuilder.addColumnDouble(value.name());
+                    break;
+                case Image:
+                    BinaryMetaData binaryMetaDataImage = new BinaryMetaData(value.name());
+                    schemaBuilder.addColumn(binaryMetaDataImage);
+                    break;
+                case Integer:
+                    schemaBuilder.addColumnInteger(value.name());
+                    break;
+                case String:
+                    schemaBuilder.addColumnString(value.name());
+                    break;
+                case Time:
+                    schemaBuilder.addColumnTime(value.name(),TimeZone.getDefault());
+                    break;
+                case Categorical:
+                    schemaBuilder.addColumnCategorical(value.name());
+                    break;
+                case Bytes:
+                    BinaryMetaData binaryMetaData = new BinaryMetaData(value.name());
+                    schemaBuilder.addColumn(binaryMetaData);
```

Comment:
We can probably combine the `Image` and `Bytes` case here

Reply:
I want to keep bytes and images separate here. Technically an ImageWritable type exists and we should figure out how we want to interact with base64/images.